### ISSUE TITLE
We need both column and row vectors

### ISSUE THREAD

**Issue by fingolfin** [gap-packages/matrices#2]
There are various reasons why I think this is necessary. For starters, it would be good to have the product of a 1xn vector times a nx1 vector to be their scalar product, and that of a nx1 vector times a 1xn a nxn matrix. 
Of course we could still allow people to multiply 1xn times 1xn and simply map this to scalar products, too...

But more importantly, the distinction becomes cruicial when working with vectors and matrices over a non-commutative domain $R$. Then row vectors are left R-module elements, while column vectors are right R-module elements (this is meant mathematically, not necessarily on the GAP level, though that might also be nice... not sure).

This corresponds to the "associative convention" (as @mohamed-barakat calls it), that is:  If $\phi$ is an R-module endomorphism, $v$ a column vector, and $a$ a scalar (i.e. an element of $R$), then they should satisfiy
  \phi(v a) = (\phi v) a
And if you want \phi to act from the right, you want
  (a v)^\phi = a v^\phi

The cruicial part here is that you multiply the matrix from the one side, the scalar from the other side. Of course you can multiple a row vector from both sides by scalars (they are R-R-bimodules), but for `MultRowVector` (and a hypothetical `MultCoumnVector`) we should make sure we multiply from the correct side (and document it as such)

Note that we may not have to do anything on the technical level to get that right; we just should keep this in mind.

So we should have IsVectorObj for common things (e.g. addition, access to the n-th entry), and then IsRowVectorObj  which is also a left module element (again: not necessarily an IsLeftModule, but mathematically)

**Comment by mohamed-barakat** [comment]
Yes, this is crucial for homalg in order to be able to deal with left and RIGHT modules over non-commutative rings. A trick would be to "transpose" which is expensive and would properly transpose would require an involution in the noncommutative ring, which might not exist!
